ceph/rados/bluestoretool_workflows.py

```diff
@@ -0,0 +1,301 @@
+        """Module to run consistency check on BlueStore metadata.
+        If deep is specified, also read all object data and verify checksums.
+        Args:
+            osd_id: OSD ID for which cbt will be executed
+        Returns:
+            Returns the output of cbt quick fsck
+        """
+        # Extracting the ceush map from the cluster
+        _cmd = "ceph-bluestore-tool fsck "
+        if deep:
+            _cmd = f"{_cmd} --deep true"
+        return self.run_cbt_command(cmd=_cmd, osd_id=osd_id)
+
+    def run_quick_consistency_check(self, osd_id: int):
+        """Module to run consistency check on BlueStore metadata comparing allocator
+         data (from RocksDB CFB when exists and if not uses allocation-file) with ONodes state.
+        Args:
+            osd_id: OSD ID for which cbt will be executed
+        Returns:
+            Returns the output of cbt quick fsck
+        """
+        _cmd = "ceph-bluestore-tool qfsck"
+        return self.run_cbt_command(cmd=_cmd, osd_id=osd_id)
+
+    def repair(self, osd_id: int):
+        """Module to run a consistency check and repair any errors it can.
+        Args:
+            osd_id: OSD ID for which cbt will be executed
+        Returns:
+            Returns the output of cbt repair cmd
+        """
+        _cmd = "ceph-bluestore-tool repair"
+        return self.run_cbt_command(cmd=_cmd, osd_id=osd_id)
+
+    def fetch_allocmap(self, osd_id: int):
+        _cmd = "ceph-bluestore-tool allocmap"
+        return self.run_cbt_command(cmd=_cmd, osd_id=osd_id)
+
+    def restore_cfb(self, osd_id: int):
+        """Module to Reverses changes done by the new NCB code
+        (either through ceph restart or when running allocmap command) and
+        restores RocksDB B Column-Family (allocator-map).
+        Args:
+            osd_id: OSD ID for which cbt will be executed
+        Returns:
+            Returns the output of cbt restore_cfb cmd
+        """
+        _cmd = "ceph-bluestore-tool restore_cfb"
+        return self.run_cbt_command(cmd=_cmd, osd_id=osd_id)
+
+    def do_bluefs_export(self, osd_id: int, output_dir: str):
+        """Module to export the contents of BlueFS (i.e., RocksDB files)
+        to an output directory
+        Args:
+            osd_id: OSD ID for which cbt will be executed
+            output_dir: Output directory for export
+        Returns:
+            Returns the output of cbt bluefs-export cmd
+        """
+        _cmd = f"ceph-bluestore-tool bluefs-export --out-dir {output_dir}"
+        return self.run_cbt_command(cmd=_cmd, osd_id=osd_id)
+
+    def block_device_sizes(self, osd_id: int):
+        """Module to print the device sizes, as understood by BlueFS, to stdout.
+        Args:
+            osd_id: OSD ID for which cbt will be executed
+        Returns:
+            Returns the output of cbt bluefs-bdev-sizes cmd
+        """
+        _cmd = "ceph-bluestore-tool bluefs-bdev-expand"
```

Comment:
I think command here needs to be fixed

Reply:
thanks for pointing out, this has been fixed